lotion/lib/abci-app.js

```diff
@@ -43,7 +43,11 @@ async function runTx(
       await txHandler(hookedState, tx, hookedChainInfo, !allowMutation);
     }
   } catch (e) {
-    return [false, e.toString()];
+    let err = e.toString();
+    if (err === '[object Object]') {
+      err = JSON.stringify(e);
```

Comment:
Agree with @johannbarbie . We should be smarter and add some error messages to the tx handlers. Also we don't know if that could be a security issue if we return the whole stringified object 🤔

Reply:
Yes, I agree with you but I was not sure how to do batter
@johannbarbie Do you think it's better just to send error instead of tx hash in case there is an error?
@pinkiebell The only other way I see it is to find out why `vmerror` is not thrown as proper exception and fix it there

Reply:
> Do you think it's better just to send error instead of tx hash in case there is an error?
yes, the user knows the txHash anyways .

What he really cares about is the moment when the tx is included.

i think the way it works in ethereum is 2 steps:
1. from `sendRawTransaction()` you get a txHash back
2. with the txHash you use `getTransactionReceipt()` to wait for the block to be mined

https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactionreceipt

i think we should follow the API, and start thinking about receipts, but outside of this bounty.

Reply:
> yes, the user knows the txHash anyways
ok, fixed that
and agree, it would be great if we can get error by calling `getTransactionReceipt`